tree/ntuple/inc/ROOT/RNTupleWriter.hxx

```diff
@@ -83,14 +118,33 @@ private:
                                                 const ROOT::RNTupleWriteOptions &options);
 
 public:
+   /// Creates an RNTupleWriter backed by `storage`, overwriting it if one with the same URI exists.
+   /// The format of the backing storage is determined by `storage`: in the simplest case it will be a local file, but
+   /// a different backend may be selected via the URI prefix. The "daos://" prefix will use the experimental DAOS
+   /// backend.
+   ///
+   /// The RNTupleWriter will create an RNTuple with the schema determined by `model` (which must not be null) and
+   /// with name `ntupleName`. This same name can later be used to read back the RNTuple via RNTupleReader.
+   ///
+   /// `options` may be passed to this function to customize the behavior of the RNTupleWriter (\see
+   /// RNTupleWriteOptions).
+   ///
    /// Throws an exception if the model is null.
    static std::unique_ptr<RNTupleWriter>
    Recreate(std::unique_ptr<ROOT::RNTupleModel> model, std::string_view ntupleName, std::string_view storage,
             const ROOT::RNTupleWriteOptions &options = ROOT::RNTupleWriteOptions());
+
+   /// Same as Recreate(std::unique_ptr<ROOT::RNTupleModel>, std::string_view, std::string_view, const
+   /// ROOT::RNTupleWriteOptions &) but the RNTupleModel will be created with the dynamically-typed fields specified in
+   /// `fields`. This overload may be used when the RNTuple schema is created at runtime with types unknown at compile
+   /// time.
```

Comment:
The recommended way should be `RFieldBase::Create` and `RNTupleModel::AddField`. This `Recreate` overload is simply a convenience function...

Reply:
I think it was implemented during a ROOT hackathon, mostly with usage from Python in mind (that was before we were able to fully use `RNTupleModel` from Python)

Reply:
FWIW, it was originally suggested [here](https://github.com/root-project/root/pull/8688#issuecomment-883396155).